js/background/background.js

```diff
@@ -674,88 +668,381 @@ class Steam {
 Steam._dynamicstore_promise = null;
 Steam._supportedCurrencies = null;
 
+class IndexedDB {
+    static init() {
+        if (!IndexedDB._promise) {
+            IndexedDB._promise = async () => {
+                IndexedDB.db = await idb.openDB("Augmented Steam", Info.db_version, {
+                    upgrade(db, oldVersion) {
+                        switch(oldVersion) {
+                            case 0: {
+                                db.createObjectStore("coupons").createIndex("appid", "appids", { unique: false, multiEntry: true });
+                                db.createObjectStore("giftsAndPasses").createIndex("appid", '', { unique: false, multiEntry: true });
+                                db.createObjectStore("items");
+                                db.createObjectStore("earlyAccessAppids");
+                                db.createObjectStore("purchases");
+                                db.createObjectStore("dynamicStore").createIndex("appid", '', { unique: false, multiEntry: true });
+                                db.createObjectStore("packages");
+                                db.createObjectStore("storePageData");
+                                db.createObjectStore("profiles");
+                                db.createObjectStore("rates");
+                                db.createObjectStore("notes");
+                                db.createObjectStore("itad");
+                                db.createObjectStore("ownedElsewhere");
+                                db.createObjectStore("lastItadSync");
+                                break;
+                            }
+                            default: {
+                                console.warn("Unknown oldVersion", oldVersion);
+                                break;
+                            }
+                        }
+                    },
+                    blocked() {
+                        console.error("Failed to upgrade database, there is already an open connection");
+                    },
+                });
+            };
+        }
+        return IndexedDB._promise;
+    }
+    static then(onDone, onCatch) {
+        return IndexedDB.init()().then(onDone, onCatch);
```

Comment:
`init()()` really? You're sure this isn't a mistake, right?

Reply:
Yes, `init` returns an `async` function, that has to be executed in order to return a `Promise`